benchmarks/sparse/dlmc/spmm.py

```diff
@@ -0,0 +1,150 @@
+# Sparse benchmarks
+
+# This benchmark is for  Sparse matrix-matrix multiplication (SPMM) performance test. 
+# They exist for comparing the performance of sparse matrix routines
+# `sparse @ sparse` and `sparse @ dense` with different backends (CPU/CUDA)
+# and with other frameworks such as scipy. 
+
+import sys
+from pathlib import Path
+import pandas as pd
+import argparse
+import torch
+import torch.utils.benchmark as benchmark_utils
+from .utils import load_spmm_dataset, to_coo_scipy, scipy_coo_matmul
+import pickle
+
+def torch_backward(a_dense, b_dense):
+    a_dense.requires_grad = True
+    b_dense.requires_grad = True
+    r1 = a_dense.matmul(b_dense)
+    f1 = torch.sum(r1)
```

Comment:
when you are doing `sum` in the function being benchmarked, it alters benchmarking results. You have to pass gradOutput to this function, and then do `r1.backward(gradOutput)`. Same for `sparse_torch_backward` below